draft/jumpstarter.rst

```diff
@@ -0,0 +1,218 @@
+.. vale off
+
+======================
+CEP XXXX: Jumpstarter
+======================
+
+:CEP: XXXX
+:Author: Micah Lyle
+:Implementation Team: Omer Katz
+:Shepherd: Omer Katz
+:Status: Draft
+:Type: Feature
+:Created: 2021-05-09
+:Last-Modified: 2021-05-09
+
+.. contents:: Table of Contents
+   :depth: 3
+   :local:
+
+Abstract
+========
+
+`Next-Gen Celery`_ aims to:
+
+1. Move Celery from the past to the present. 
+
+2. Resolve long-standing design bugs in our implementation. 
+
+3. Modernize the code to support Python3+. 
```

Comment:
python 3.6/3.7+?

Reply:
We don't know yet but it's likely to be 3.7+ or even 3.10+ (in case we want to use pattern matching).

Reply:
3.10+ may also have some niceities with async libraries although I don't think exception groups is landing in 3.10 (https://www.python.org/dev/peps/pep-0654/)

Reply:
That will probably hit 3.11 or higher and it will take time for Trio and AnyIO to adapt.

Reply:
Another nice thing about `3.10` in my opinion is that it makes:

``from __future__ import annotation`` completely optional, and 3.10 has a number of type-hinting improvements.

How important do you think being able to have a nicely type-hinted codebase is, vs. supporting 3.7/3.8/3.9+? We could do the `typing_extensions` workaround. But I've really loved seeing @thedrow's jumpstarter code with the very modern type hints, including `|` usage, self-referential annotations, etc. I feel like 3.10 is a pretty big milestone with that.

All of that being said though, this can all be emulated with the `__future__` import and `typing_extensions` I think.